test.sh

```diff
@@ -18,7 +18,7 @@ fi
 mv download-new.sh download.sh
 if [ ! -d "downloads" ]; then
   echo "Downloading all files into the cache (./downloads/)"
-  mkdir downloads
+  mkdir -p downloads
```

Comment:
It shouldn't be possible for this line to be reached if there is an existing downloads folder, because of the test above:

`if [ ! -d "downloads" ]; then` ~= if the folder "downloads" doesn't exist.

But having the -p shouldn't hurt anything :)

Reply:
Don't ask me, my MAC was somehow able to reach the line with an existing downloads folder and I just put the -p then : )

Reply:
If you throw in the text that was output when the error happened into a comment, we can see if there is a workaround.

Reply:
alright, so I removed the `-p` on my mac and when I first start the `./test.sh` I get:

```bash
Starting a test run. Working data and environments will be placed in ./auto-test/

Looking for: ['nodejs']

pkgs/main/osx-64                                     5.0MB @   3.0MB/s  1.7s
bioconda/noarch                                      4.6MB @   2.7MB/s  1.7s
pkgs/r/noarch                                                 No change
pkgs/r/osx-64                                                 No change
pkgs/main/noarch                                              No change
bioconda/osx-64                                      4.2MB @   2.1MB/s  2.0s
conda-forge/noarch                                  12.0MB @   3.9MB/s  3.1s
conda-forge/osx-64                                  26.6MB @   5.4MB/s  4.9s
Transaction

  Prefix: /Users/martin/git/2023-Workshop-MISSIoN-Nanopore/.codedown

  Updating specs:

   - nodejs


  Package              Version  Build       Channel           Size
────────────────────────────────────────────────────────────────────
  Install:
────────────────────────────────────────────────────────────────────

  + icu                   72.1  h7336db1_0  conda-forge     Cached
  + ca-certificates  2023.7.22  h8857fd0_0  conda-forge     Cached
  + libuv               1.44.2  h0dc2134_1  conda-forge     Cached
  + libzlib             1.2.13  h8a1eda9_5  conda-forge     Cached
  + libcxx              16.0.6  hd57cbcb_0  conda-forge     Cached
  + openssl              3.1.2  h8a1eda9_0  conda-forge     Cached
  + zlib                1.2.13  h8a1eda9_5  conda-forge     Cached
  + nodejs              20.1.0  h46e3395_0  conda-forge     Cached

  Summary:

  Install: 8 packages

  Total download: 0 B

────────────────────────────────────────────────────────────────────



Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

To activate this environment, use

     $ mamba activate /Users/martin/git/2023-Workshop-MISSIoN-Nanopore/.codedown

To deactivate an active environment, use

     $ mamba deactivate

dyld[3225]: Library not loaded: @rpath/libuv.1.dylib
  Referenced from: <624E9477-EF83-3CF3-8702-508405D10F47> /Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/lib/libnode.115.dylib
  Reason: tried: '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/usr/local/lib/libuv.1.dylib' (no such file), '/usr/lib/libuv.1.dylib' (no such file, not in dyld cache)
./test.sh: line 10:  3225 Abort trap: 6           npm install -g codedown
./test.sh: line 13: codedown: command not found
xargs: cat: terminated with signal 13; aborting
The set of files to download has changed. Invalidating the cache (./downloads/)
Downloading all files into the cache (./downloads/)
/Users/martin/git/2023-Workshop-MISSIoN-Nanopore
cp: ../downloads/*: No such file or directory
./test.sh: line 37: codedown: command not found
xargs: cat: terminated with signal 13; aborting
```

Then, when i rerun this it activates the env and starts the downloads, which take a bit. Then it breaks with:

```
uniprot_sprot.fasta.gz                        100%[===============================================================================================>]  87,50M   166KB/s    in 7m 58s

2023-08-15 19:53:55 (187 KB/s) - ‘uniprot_sprot.fasta.gz’ saved [91753753]

/Users/martin/git/2023-Workshop-MISSIoN-Nanopore
+ mkdir envs
+ mamba create -y -p envs/workshop fastqc nanoplot filtlong flye bandage minimap2 tablet racon samtools igv

Looking for: ['fastqc', 'nanoplot', 'filtlong', 'flye', 'bandage', 'minimap2', 'tablet', 'racon', 'samtools', 'igv']

conda-forge/osx-64                                          Using cache
conda-forge/noarch                                          Using cache
bioconda/osx-64                                             Using cache
bioconda/noarch                                             Using cache
pkgs/r/noarch                                                 No change
pkgs/r/osx-64                                                 No change
pkgs/main/osx-64                                              No change
pkgs/main/noarch                                              No change
Could not solve for environment specs
The following package could not be installed
└─ bandage does not exist (perhaps a typo or a missing channel).
```

okay, maybe a mac issue... 

I rerun again and now I get:

```bash
To activate this environment, use

     $ mamba activate /Users/martin/git/2023-Workshop-MISSIoN-Nanopore/.codedown

To deactivate an active environment, use

     $ mamba deactivate

dyld[3818]: Library not loaded: @rpath/libuv.1.dylib
  Referenced from: <624E9477-EF83-3CF3-8702-508405D10F47> /Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/lib/libnode.115.dylib
  Reason: tried: '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/Users/martin/miniconda3/pkgs/nodejs-20.1.0-h46e3395_0/bin/../lib/libuv.1.dylib' (no such file), '/usr/local/lib/libuv.1.dylib' (no such file), '/usr/lib/libuv.1.dylib' (no such file, not in dyld cache)
./test.sh: line 10:  3818 Abort trap: 6           npm install -g codedown
./test.sh: line 13: codedown: command not found
xargs: cat: terminated with signal 13; aborting
The set of files to download has changed. Invalidating the cache (./downloads/)
Downloading all files into the cache (./downloads/)
/Users/martin/git/2023-Workshop-MISSIoN-Nanopore
cp: ../downloads/*: No such file or directory
./test.sh: line 37: codedown: command not found
xargs: cat: terminated with signal 13; aborting
```

Honestly, not sure if it makes sense solving this on a MAC with M2 chip... as written on Matrix I am on it testing it on my system76 with fresh mamba install